tests/queries/0_stateless/02169_map_functions.sql

```diff
@@ -0,0 +1,10 @@
+DROP TABLE IF EXISTS table_map;
```

Comment:
It's defensively need more tests

`select mapMap((x, y) -> (x, x + 1), map(1, 0, 2, 0))`

```
2022.02.09 18:10:04.947759 [ 621942 ] {3b6c7789-a520-4903-a778-bb1e0feeb233} <Fatal> : Logical error: 'Bad cast from type DB::ColumnVector<unsigned short> to DB::ColumnVector<char8_t>'.
2022.02.09 18:10:04.949136 [ 622146 ] {} <Fatal> BaseDaemon: ########################################
2022.02.09 18:10:04.949638 [ 622146 ] {} <Fatal> BaseDaemon: (version 22.2.1.3217, build id: 30CF9EEB87BD26CE) (from thread 621942) (query_id: 3b6c7789-a520-4903-a778-bb1e0feeb233) Received signal Aborted (6)
2022.02.09 18:10:04.950025 [ 622146 ] {} <Fatal> BaseDaemon: 
2022.02.09 18:10:04.950347 [ 622146 ] {} <Fatal> BaseDaemon: Stack trace: 0x7f912b27918b 0x7f912b258859 0x1651fcf8 0x1651fe02 0x1f3000c4 0x234e37c5 0x23494222 0x234d1a3f 0x234f7a32 0x234a4805 0x234d0bf7 0x251a9cfb 0x251a9a6d 0x25163e8f 0x25161e5f 0x2515bce0 0x2516baa5 0x2937e779 0x2937ef88 0x295c24d4 0x295befba 0x295bdd9c 0x7f912b42e609 0x7f912b355293
2022.02.09 18:10:04.950746 [ 622146 ] {} <Fatal> BaseDaemon: 4. gsignal @ 0x7f912b27918b in ?
2022.02.09 18:10:04.950840 [ 622146 ] {} <Fatal> BaseDaemon: 5. abort @ 0x7f912b258859 in ?
2022.02.09 18:10:05.016329 [ 622146 ] {} <Fatal> BaseDaemon: 6. ./obj-x86_64-linux-gnu/../src/Common/Exception.cpp:52: DB::handle_error_code(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool, std::__1::vector<void*, std::__1::allocator<void*> > const&) @ 0x1651fcf8 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:05.070102 [ 622146 ] {} <Fatal> BaseDaemon: 7. ./obj-x86_64-linux-gnu/../src/Common/Exception.cpp:59: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ 0x1651fe02 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:05.452432 [ 622146 ] {} <Fatal> BaseDaemon: 8. ./obj-x86_64-linux-gnu/../src/Common/typeid_cast.h:40: std::__1::enable_if<std::is_reference_v<DB::ColumnVector<char8_t> const&>, DB::ColumnVector<char8_t> const&>::type typeid_cast<DB::ColumnVector<char8_t> const&, DB::IColumn const>(DB::IColumn const&) @ 0x1f3000c4 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:05.558118 [ 622146 ] {} <Fatal> BaseDaemon: 9. ./obj-x86_64-linux-gnu/../src/DataTypes/Serializations/SerializationNumber.cpp:138: DB::SerializationNumber<char8_t>::serializeBinaryBulk(DB::IColumn const&, DB::WriteBuffer&, unsigned long, unsigned long) const @ 0x234e37c5 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:05.629800 [ 622146 ] {} <Fatal> BaseDaemon: 10. ./obj-x86_64-linux-gnu/../src/DataTypes/Serializations/ISerialization.cpp:115: DB::ISerialization::serializeBinaryBulkWithMultipleStreams(DB::IColumn const&, unsigned long, unsigned long, DB::ISerialization::SerializeBinaryBulkSettings&, std::__1::shared_ptr<DB::ISerialization::SerializeBinaryBulkState>&) const @ 0x23494222 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:05.689933 [ 622146 ] {} <Fatal> BaseDaemon: 11. ./obj-x86_64-linux-gnu/../src/DataTypes/Serializations/SerializationNamed.cpp:55: DB::SerializationNamed::serializeBinaryBulkWithMultipleStreams(DB::IColumn const&, unsigned long, unsigned long, DB::ISerialization::SerializeBinaryBulkSettings&, std::__1::shared_ptr<DB::ISerialization::SerializeBinaryBulkState>&) const @ 0x234d1a3f in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:05.793952 [ 622146 ] {} <Fatal> BaseDaemon: 12. ./obj-x86_64-linux-gnu/../src/DataTypes/Serializations/SerializationTuple.cpp:365: DB::SerializationTuple::serializeBinaryBulkWithMultipleStreams(DB::IColumn const&, unsigned long, unsigned long, DB::ISerialization::SerializeBinaryBulkSettings&, std::__1::shared_ptr<DB::ISerialization::SerializeBinaryBulkState>&) const @ 0x234f7a32 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:05.904989 [ 622146 ] {} <Fatal> BaseDaemon: 13. ./obj-x86_64-linux-gnu/../src/DataTypes/Serializations/SerializationArray.cpp:309: DB::SerializationArray::serializeBinaryBulkWithMultipleStreams(DB::IColumn const&, unsigned long, unsigned long, DB::ISerialization::SerializeBinaryBulkSettings&, std::__1::shared_ptr<DB::ISerialization::SerializeBinaryBulkState>&) const @ 0x234a4805 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:05.993245 [ 622146 ] {} <Fatal> BaseDaemon: 14. ./obj-x86_64-linux-gnu/../src/DataTypes/Serializations/SerializationMap.cpp:296: DB::SerializationMap::serializeBinaryBulkWithMultipleStreams(DB::IColumn const&, unsigned long, unsigned long, DB::ISerialization::SerializeBinaryBulkSettings&, std::__1::shared_ptr<DB::ISerialization::SerializeBinaryBulkState>&) const @ 0x234d0bf7 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:06.094454 [ 622146 ] {} <Fatal> BaseDaemon: 15. ./obj-x86_64-linux-gnu/../src/Formats/NativeWriter.cpp:63: DB::writeData(DB::ISerialization const&, COW<DB::IColumn>::immutable_ptr<DB::IColumn> const&, DB::WriteBuffer&, unsigned long, unsigned long) @ 0x251a9cfb in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:06.187416 [ 622146 ] {} <Fatal> BaseDaemon: 16. ./obj-x86_64-linux-gnu/../src/Formats/NativeWriter.cpp:164: DB::NativeWriter::write(DB::Block const&) @ 0x251a9a6d in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:06.498413 [ 622146 ] {} <Fatal> BaseDaemon: 17. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:1792: DB::TCPHandler::sendData(DB::Block const&) @ 0x25163e8f in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:06.804538 [ 622146 ] {} <Fatal> BaseDaemon: 18. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:706: DB::TCPHandler::processOrdinaryQueryWithProcessors() @ 0x25161e5f in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:07.076228 [ 622146 ] {} <Fatal> BaseDaemon: 19. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:346: DB::TCPHandler::runImpl() @ 0x2515bce0 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:07.386005 [ 622146 ] {} <Fatal> BaseDaemon: 20. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:1909: DB::TCPHandler::run() @ 0x2516baa5 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:07.449458 [ 622146 ] {} <Fatal> BaseDaemon: 21. ./obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerConnection.cpp:43: Poco::Net::TCPServerConnection::start() @ 0x2937e779 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:07.524620 [ 622146 ] {} <Fatal> BaseDaemon: 22. ./obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerDispatcher.cpp:115: Poco::Net::TCPServerDispatcher::run() @ 0x2937ef88 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:07.604328 [ 622146 ] {} <Fatal> BaseDaemon: 23. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/ThreadPool.cpp:199: Poco::PooledThread::run() @ 0x295c24d4 in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:07.683264 [ 622146 ] {} <Fatal> BaseDaemon: 24. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Thread.cpp:56: Poco::(anonymous namespace)::RunnableHolder::run() @ 0x295befba in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:07.761267 [ 622146 ] {} <Fatal> BaseDaemon: 25. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Thread_POSIX.cpp:345: Poco::ThreadImpl::runnableEntry(void*) @ 0x295bdd9c in /home/ubuntu/_tmp/clickhouse
2022.02.09 18:10:07.761544 [ 622146 ] {} <Fatal> BaseDaemon: 26. ? @ 0x7f912b42e609 in ?
2022.02.09 18:10:07.761788 [ 622146 ] {} <Fatal> BaseDaemon: 27. clone @ 0x7f912b355293 in ?
2022.02.09 18:10:08.819656 [ 622146 ] {} <Fatal> BaseDaemon: Calculated checksum of the binary: EB970097569645DD421D79F24D19B235. There is no information about the reference checksum.

```